SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -97,6 +98,43 @@ void CIncNSSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container
 
   ComputeVorticityAndStrainMag<1>(*config, iMesh);
 
+  /*--- Compute recovered pressure and temperature for streamwise periodic flow ---*/
+  if (config->GetKind_Streamwise_Periodic() != NONE) {
+
+    /*--- Define and initialize helping variables ---*/
+    su2double dot_product, Pressure_Recovered, Temperature_Recovered;
+
+    const su2double delta_p  = config->GetStreamwise_Periodic_PressureDrop() / config->GetPressure_Ref();
+
+    /*--- Reference node on inlet periodic marker to compute relative distance along periodic translation vector. ---*/
+    const su2double* ReferenceNode = geometry->GetStreamwise_Periodic_RefNode();
+
+    /*--- Compute square of the distance between the 2 periodic surfaces. ---*/
+    const su2double norm2_translation = GeometryToolbox::SquaredNorm(nDim, config->GetPeriodic_Translation(0));
+
+    /*--- Compute recoverd pressure and temperature for all points ---*/
+    for (auto iPoint = 0ul; iPoint < nPoint; iPoint++) {
```

Comment:
SU2_OMP_FOR_STAT(omp_chunk_size) on top

Reply:
Done (yeah i should have looked more carefully at those openMP things 😞 )